### DISCUSSION TITLE
Getting repo public-key not possible with GITHUB_TOKEN?

### DISCUSSION THREAD

**Discussion by biatwc** [community/community#24926] [Actions]
<p>Hello</p>
<p>I try to use ${{ secrets.GITHUB_TOKEN }} to pull the public-key from the same repository as the workflow runs on and I get:</p>
<p>Resource not accessible by integration.</p>
<p>The documentation <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token" class="inline-onebox" rel="noopener nofollow ugc">Authentication in a workflow - GitHub Docs</a> lists the permissions and it does apparently let that token pull the public-key.</p>
<p>Anyone know what is going on?</p>
<p>Code below:</p>
<pre><code class="lang-auto">name: Test token
on:
  workflow_dispatch:
jobs:
   get_token:
    name: "Get and set Github token"
    runs-on: [self-hosted,live]
    steps:
    - name: Get public-key
      run: |
        curl -X GET --url https://api.github.com/repos/${{ github.repository }}/actions/secrets/public-key --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' --header 'Accept: application/vnd.github.v3+json'
      shell: bash
</code></pre>
<p>Results in:</p>
<pre><code class="lang-auto">  curl -X GET --url https://api.github.com/repos/myorg/myrepo/actions/secrets/public-key --header 'authorization: ***' --header 'Accept: application/vnd.github.v3+json'
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   159  100   159    0     0    524      0 --:--:-- --:--:-- --:--:--   524
{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/reference/actions#get-a-repository-public-key"
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>The documentation for the API endpoint you’re using (<a href="https://docs.github.com/en/rest/reference/actions#get-a-repository-public-key">Get a repository public key</a>) says this:</p>
<blockquote>
<p>GitHub Apps must have the <code>secrets</code> repository permission to use this endpoint.</p>
</blockquote>
<p><code>secrets</code> is not in the list of permissions for the <code>GITHUB_TOKEN</code>.</p>

**Comment by biatwc** [comment]
<p>Thanks for your answer.<br>
I find it confusing this endpoint is listed under the contents permissions though.</p>
